src/adaXT/random_forest/RandomForest.py

```diff
@@ -0,0 +1,355 @@
+        # take care of copying everything in the correct format
+        res[:] = array[:]
+
+    def read(self):
+        '''
+        Reads the array from the shared memory without unnecessary copying.
+        '''
+        # simply create an array of the correct shape and type,
+        # using the shared memory location we created earlier
+        return np.ndarray(self._shape, self._dtype, buffer=self._shared.buf)
+
+    def unlink(self):
+        '''
+        Releases the allocated memory. Call when finished using the data,
+        or when the data was copied somewhere else.
+        '''
+        self._shared.close()
+        self._shared.unlink()
+
+
+class RandomForest:
+    '''
+    The Random Forrest
+    '''
+
+    def __init__(
+            self,
+            forrest_type: str,
+            n_estimators: int = 100,
+            criterion: Criteria = Squared_error,
+            bootstrap: bool = True,
+            n_jobs: int = 1,
+            max_samples: int = None,
+            max_features: None = None,
+            max_depth: int = sys.maxsize,
+            impurity_tol: float = 0,
+            min_samples_split: int = 1,
+            min_samples_leaf: int = 1,
+            min_improvement: float = 0,
+            splitter: Splitter | None = None):
+        """
+        Parameters
+        ----------
+        forrest_type : str
+            Classification or Regression
+        n_estimators : int, default=100
+            The number of trees in the forest.
+        criterion : Criteria, default=Squared_errror
+            The criteria function used to evaluate a split
+        bootstrap : bool, default=True
+            Whether bootstrap is used when building trees
+        n_jobs : int, default=1
+            The number of processes used to fit, and predict for the forest, -1 uses all available proccesors
+        max_samples : int, default=None
+            The number of samples drawn when doing bootstrap
+        max_features: int, float, {"sqrt", "log2"}, default=None
+            The number of features used when doing feature-sampling
+        max_depth : int
+            maximum depth of the tree, by default maximum system size
+        impurity_tol : float
+            the tolerance of impurity in a leaf node, by default 0
+        min_samples_split : int
+            the minimum amount of samples in a split, by default 1
+        min_samples_leaf : int
+            the minimum amount of samples in a leaf node, by default 1
+        min_improvement: float
+            the minimum improvement gained from performing a split, by default 0
+        splitter : Splitter | None, optional
+            Splitter class if None uses premade Splitter class
+        """
+
+        self.forrest_type = forrest_type
+        self.n_estimators = n_estimators
+        self.criterion = criterion
+        self.bootstrap = bootstrap
+        self.n_jobs = n_jobs if n_jobs != -1 else cpu_count()
```

Comment:
Would it perhaps be sensible to make sure that n_jobs is not larger
than the cpu_count(). Generally well known that having more jobs than 
logical cores leads to worse overall performance.

Reply:
I think it is fine to give the user the responsibility to not choose too many jobs, since there is the option to simply use -1.